datumaro/datumaro/components/project.py

```diff
@@ -315,6 +315,46 @@ def categories(self):
         return self._parent.categories()
 
 class Dataset(Extractor):
+    @classmethod
+    def from_iterable(cls, iterable, categories=None):
+        """Generation of Dataset from iterable object
+
+        Args:
+            iterable: Iterable object contains DatasetItems
+            categories (dict, optional): Dict of categories of datasets. Defaults to {}.
+
+        Returns:
+            Dataset: Dataset object
+        """
+
+        # Configuration of categories
+        if not categories:
+            categories = {}
+
+        # Get Dataset with specified categories
+        dataset = Dataset(categories=categories)
+
+        #Get iterator
+        it = iter(iterable)
+
+        # Merging items
+        subsets = defaultdict(lambda: Subset(dataset))
+
+        for item in it:
+                existing_item = subsets[item.subset].items.get(item.id)
```

Comment:
I suggest to think about extracting common code. Actually, this function can be implemented with `from_extractors` by creating a simple `Extractor` subclass here.